main.py

```diff
@@ -0,0 +1,241 @@
+    --db=database_name Database to save session data
+ """
+
+import argparse
+import cmd
+import os
+import sys
+
+
+from colorama import *
+from docopt import docopt, DocoptExit
+from pyfiglet import figlet_format
+from termcolor import cprint
+
+from models.amity import Amity
+
+init(wrap=True)
+
+
+def docopt_cmd(func):
+    def fn(self, arg):
+        try:
+            opt = docopt(fn.__doc__, arg, version='Amity 1.0')
+
+        except DocoptExit as error:
+
+            # The DocoptExit is thrown when the args do not match
+            # We print a message to the user and the usage block
+            print("Invalid Command!")
+            print(error)
+            return
+
+        except SystemExit:
+            # The SystemExit exception prints the usage for --help
+            # We do not need to do the print here
+            return
+
+        return func(self, opt)
+
+    fn.__name__ = func.__name__
+    fn.__doc__ = func.__doc__
+    fn.__dict__.update(func.__dict__)
+
+    return fn
+
+
+class AmityApp(cmd.Cmd):
+    cprint("\n")
+    cprint(figlet_format("AMITY ALLOCATION".center(
+        15), font="standard"), "yellow", attrs=["bold"])
+
+    def introduction():
+        print (""" Amity Commands
+        Usage:
+            Amity create_room (Office|Living-Space|Space) (Male|m|Female|f|None|n) <room_name>...
+            Amity list_rooms
+            Amity delete_room <room_identifier>
+            Amity add_person <person_name> <person_gender> (Fellow|Staff) [<wants_accommodation>]
+            Amity list_people
+            Amity delete_person <person_identifier>
+            Amity reallocate_person <person_identifier> <new_room_name>
+            Amity load_people
+            Amity print_allocations [-o=<filename>]
+            Amity print_unallocated [-o=<filename>]
+            Amity print_room <room_name>
+            Amity save_state [--db=<database_name>]
+            Amity load_state [--db=<database_name>]
+            Amity (-i | --interactive)
+            Amity (-h | --help)
+            Amity (--version)
+        Options:
+            -i, --interactive   Interactive Mode
+            -h, --help Show on this screen and then exit
+            --version   Show version and exit
+            -o=filename]    Specify output file
+            --db=database_name Database to save session data
+        """)
+        cprint("\n")
+
+    intro = introduction()
+    prompt = "(Amity)>> "
+    file = None
+
+    @docopt_cmd
+    def do_create_room(self, args):
+        """Usage: create_room (Office|Living-Space|Space) (Male|m|Female|f|None|n) <room_name>..."""
+
+        if args['Office']:
+            room_type = 'office'
+        elif args['Living-Space'] or args['Space']:
+            room_type = 'space'
+
+        if args['Male'] or args['m']:
+            room_gender = 'M'
+        elif args['Female'] or args['f']:
+            room_gender = 'F'
+        elif args['None'] or args['n']:
```

Comment:
does the app allow a person to be allocated a room if the gender is not specified ? and if so why have it in the first place? kinda beats the logic. plus as far as i'm concerned you are either male or female so I don't understand why someone can not specify the gender